app/src/main/java/com/sdp/movemeet/utility/ImageHandler.java

```diff
@@ -0,0 +1,71 @@
+package com.sdp.movemeet.utility;
+
+import android.net.Uri;
+import android.util.Log;
+import android.view.View;
+import android.widget.ProgressBar;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.storage.UploadTask;
+import com.sdp.movemeet.backend.BackendManager;
+import com.sdp.movemeet.backend.firebase.storage.StorageImageManager;
+import com.sdp.movemeet.models.Image;
+import com.squareup.picasso.Picasso;
+
+public class ImageHandler {
+
+    private static final String TAG = "FirebaseInteraction";
+
+    private static BackendManager<Image> imageBackendManager;
```

Comment:
I wonder if we should define this as BackendManager<Image>, or directly as StorageImageManager ? I guess the first option is best in term of abstraction, but the second prevent useless casts.

Reply:
Well, I'm not sure as well, but this is to stay consistent with the way we implemented all our different kinds of managers so far (such as: `userManager`, `activityBackendManager`, `messageManager`, etc.). I think Kepler would have done it this way as well! This way of defining the different backend managers really highlights our abstraction I guess.